ldelf/ta_elf.c

```diff
@@ -247,10 +257,20 @@ static void e32_save_symtab(struct ta_elf *elf, size_t tab_idx)
 	size_t str_idx = shdr[tab_idx].sh_link;
 
 	elf->dynsymtab = (void *)(shdr[tab_idx].sh_addr + elf->load_addr);
-	assert(!(shdr[tab_idx].sh_size % sizeof(Elf32_Sym)));
+	if (!ALIGNMENT_IS_OK(elf->dynsymtab, Elf32_Sym))
+		err(TEE_ERROR_GENERIC, "Bad alignment of dynsymtab %p",
+		    elf->dynsymtab);
+	check_range(elf, "dynsymtab", elf->dynsymtab, shdr[tab_idx].sh_size);
+
+	if (shdr[tab_idx].sh_size % sizeof(Elf32_Sym))
+		err(TEE_ERROR_GENERIC,
+		    "Size of dynsymtab not an even multiple of Elf32_Sym");
 	elf->num_dynsyms = shdr[tab_idx].sh_size / sizeof(Elf32_Sym);
 
+	if (str_idx >= elf->e_shnum)
+		err(TEE_ERROR_GENERIC, "Dynstr section index out of range");
 	elf->dynstr = (void *)(shdr[str_idx].sh_addr + elf->load_addr);
+	check_range(elf, "dynstr", elf->dynstr, shdr[str_idx].sh_size);
 	elf->dynstr_size = shdr[str_idx].sh_size;
```

Comment:
I suggest an empty line above line 273 (`check_range(...);`).

Reply:
I ended up adding one after instead, hope that's OK.